spec/basket_spec.rb

```diff
@@ -2,19 +2,35 @@
 
 RSpec.describe Basket do
 
-  it "creates basket" do
-    expect Basket.new.is_a? Basket
-  end
+  let(:product1) { Product.new(name: "Football", price: 79) }
+  let(:product2) { Product.new(name: "Basketball", price: 60) }
+  let(:basket) { Basket.new([product1, product2]) }
 
-  let(:p1) { Product.new(name: "Football", price: 79) }
-  let(:p2) { Product.new(name: "Basketball", price: 60) }
-  it "checks if sum with vat is correct" do
-    sum = Basket.new([p1, p2]).sum
-    sum_with_vat = Basket.new([p1, p2]).sum_with_vat
-    expect(sum * 1.23).to eq sum_with_vat
+  describe "#sum" do
+    it "calculates products sum correctly" do
+      expect(basket.sum).to eq 139
+    end
   end
 
+  describe "#sum_with_vat" do
+    it "checks if sum with vat is correct" do
+      expect(basket.sum_with_vat).to eq 170.97
+    end
+  end
 
+  describe "#delete" do
+    context "when product does not exist in basket" do 
+      it "raises ProductNotFound error" do
+        expect {
+          basket.delete(0)
+        }.to raise_error(Basket::ProductNotFound)
+      end
+    end
 
-  #it "doesn't take basket arguments"
-end
\ No newline at end of file
+    context "when product exists in basket" do
+      it "deletes product from basket" do
+        expect(basket.delete(product1.id)). to eq (product1)
```

Comment:
sometimes you are using `eq(foo)` and another time `eq foo`. Be consistent and use the first version ;)